core_impl/src/main/java/com/google/instrumentation/trace/TimeConverter.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2017, Google Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.instrumentation.trace;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.instrumentation.common.Timestamp;
+import com.google.instrumentation.common.TimestampFactory;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * This class provides a mechanism for converting {@link Time} values to {@link Timestamp}. An
+ * instance of this class in initialized with a pair of timestamp/nanoTime measurements that wre
+ * taken at approximately the same time.
```

Comment:
This second sentence is implementation detail, not needed for usage of the class, suggest deleting it, and moving it to a // comment somewhere.

Also, you might want to replace "at approximately the" with "as close as possible to"